laughter/laughter_classification/train_rnn_laugh_classifier.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python
+import os
+import click
+
+from sklearn.model_selection import train_test_split
+import pandas as pd
+import torch.optim as optim
+import torch.nn.functional as F
+
+from laughter_classification.rnn_laugh_classifier import *
+
+
+CORPUS_ROOT = 'vocalizationcorpus'
+FRAME_LEN_SEC = 0.1
+N_MFCC = 20
+N_MEL = 128
+N_MFCC_HID = 10
+N_MEL_HID = 50
+N_EPOCHS = 50
+LR = 0.01
+SEED = 117
+
+
+def generate_data(dataset_file):
+    from laughter_classification.sspnet_data_sampler import SSPNetDataSampler
+    s = SSPNetDataSampler(CORPUS_ROOT)
+    return s.create_sampled_df(FRAME_LEN_SEC, save_path=dataset_file,
+                               force_save=True)
+
+
+def prepare_dataset(dataset):
+    snames = dataset.SNAME.unique()
+    train_snames, test_snames = train_test_split(snames, random_state=SEED)
+    xs_train = [dataset[dataset.SNAME == sname].iloc[:, :-2].values for sname in train_snames]
```

Comment:
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.isin.html

Reply:
I don't see how `DataFrame.isin()` woud help here. I don't need to just find all rows with `SNAME` from `train_snames`, I also want to split the dataframe into training objects by `SNAME`.